### ISSUE TITLE
Syntax highlighting for non-R code

### ISSUE THREAD

**Issue by boyercb** [rstudio/distill#434]
Thanks so much for  `distill` it's great!

I am writing a post and want to include some Stan code, but it's currently not caught by the syntax highlighter. I thought maybe `downlit` was an issue so I tried disabling via the `highlight_downlit` option to no avail? Is it possible to highlight non-R code?

**Comment by cderv** [comment]
Hi, 

Thanks for the report. 

I believe this is an issue because **distill** relies on Pandoc for its highlighting and Pandoc does not recognize stan language chunk. 
```powershell
❯ pandoc --list-highlight-languages | grep stan
````
The file stan.xml is not in https://github.com/jgm/skylighting/tree/master/skylighting-core/xml but it is upstream https://github.com/KDE/syntax-highlighting/tree/master/data/syntax so I guess it could be added easily. However, this would only be available in later version of Pandoc in that case. 

A possible workaround would be to use a Pandoc argument to provide custom syntax definition file (https://pandoc.org/MANUAL.html#syntax-highlighting)

````markdown
---
title: "Test Title"
author: Test author
date: 2021-12-31
output:
  distill::distill_article:
    pandoc_args: ["--syntax-definition", "stan.xml"]
---

```{r, include = FALSE}
# download the stan file
xfun::download_file("https://raw.githubusercontent.com/KDE/syntax-highlighting/master/data/syntax/stan.xml")
```

``` stan
parameters {
  real y[2];
}
model {
  y[1] ~ normal(0, 1);
  y[2] ~ double_exponential(0, 2);
}
```

````

Not ideal, but it does the job. Does it work for you ? 

For the context, **downlit** is only use to process R code chunk for better highlighting parser than Pandoc.

**Comment by boyercb** [comment]
Thanks! This worked for me.

**Comment by cderv** [comment]
Great ! Glad it works for you ! 

I leave that open as a reminder to PR into Pandoc for adding the stan.xml file there. I don't think it is wise for **rmarkdown** to duplicate those file. 

We'll try to document this also somewhere. (One of the book or at least a blog post)

**Comment by cderv** [comment]
FWIW `stan.xml` is now included in Pandoc (since 2.17.0.1 I believe)
